Python/ceval.c

```diff
@@ -740,14 +742,24 @@ _PyEval_MatchClass(PyThreadState *tstate, PyObject *subject, PyObject *type,
     if (PyObject_IsInstance(subject, type) <= 0) {
         return NULL;
     }
+    // Short circuit if there aren't any arguments:
+    Py_ssize_t nkwargs = PyTuple_GET_SIZE(kwargs);
+    Py_ssize_clean_t nattrs = nargs + nkwargs;
+    if (!nattrs) {
+        PyObject *attrs = PyTuple_New(0);
+        return attrs;
+    }
     // So far so good:
-    PyObject *seen = PySet_New(NULL);
-    if (seen == NULL) {
-        return NULL;
+    PyObject *seen = NULL;
+    if (nattrs > 1) {
```

Comment:
```suggestion
    if (nargs > 1 || (nargs == 1 && nkwargs > 0)) {
```
As mentioned in https://github.com/python/cpython/issues/138912, I'd also like to change this. If there are only keyword arguments, it's easy to see if two are the same name. Furthermore linters are able to detect these based on the AST alone.

It would also be great to say if we could suggest users to prefer keyword arguments. Not only are they more explicit, they are also faster. (No lookup of `__match_args__` and, with the change, no expensive duplicate name detection.)

Reply:
Added this in https://github.com/python/cpython/pull/138915/commits/f5852537a019b225d3586ef1806effe8e03913c6 with `if (nargs > 0 && nattrs > 1)`. I can revert it if necessary.

Reply:
Only mentioned it on the [issue](https://github.com/python/cpython/issues/138912#issuecomment-3294228864) a while ago but the keyword attributes are validated already during the compile stage at which point a SyntaxError is raised. Just added an explicit test case for it in https://github.com/python/cpython/pull/138915/commits/9375e2b27e27ce6ce5ca28128b0403eb894d3553.